src/core/HorizontalTabs/HorizontalTabs.tsx

```diff
@@ -112,24 +108,35 @@ export const HorizontalTabs = (props: HorizontalTabsProps) => {
   const [tablistSizeRef] = useResizeObserver(updateTabsWidth);
   const refs = useMergedRefs(tablistRef, tablistSizeRef);
 
-  const [currentIndex, setCurrentIndex] = React.useState(getValidIndex());
-  const [stripeStyle, setStripeStyle] = React.useState<React.CSSProperties>({});
-
+  const [currentActiveIndex, setCurrentActiveIndex] = React.useState(() =>
+    activeIndex != null
+      ? getBoundedValue(activeIndex, 0, labels.length - 1)
+      : 0,
+  );
   React.useLayoutEffect(() => {
-    if (activeIndex != null && currentIndex !== activeIndex) {
-      setCurrentIndex(getValidIndex());
+    if (activeIndex != null && currentActiveIndex !== activeIndex) {
+      setCurrentActiveIndex(getBoundedValue(activeIndex, 0, labels.length - 1));
     }
-  }, [activeIndex, currentIndex, getValidIndex]);
+  }, [activeIndex, currentActiveIndex, labels.length]);
 
+  const [stripeStyle, setStripeStyle] = React.useState<React.CSSProperties>({});
```

Comment:
move this up, close to other vars of useState, diff wuld also be nicer shown

Reply:
I moved it down closer to the useEffect which uses this state

Reply:
stripeStyle looks still randomly placed to me...

Reply:
ah, you put there because of setter..well..ok